streams/src/test/java/org/apache/kafka/streams/tests/StreamsStandByReplicaTest.java

```diff
@@ -0,0 +1,224 @@
+
+        final Serde<String> stringSerde = Serdes.String();
+
+        final Properties streamsProperties = new Properties();
+        streamsProperties.put(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, kafka);
+        streamsProperties.put(StreamsConfig.APPLICATION_ID_CONFIG, "kafka-streams-standby-tasks");
+        streamsProperties.put(StreamsConfig.DEFAULT_KEY_SERDE_CLASS_CONFIG, Serdes.String().getClass());
+        streamsProperties.put(StreamsConfig.DEFAULT_VALUE_SERDE_CLASS_CONFIG, Serdes.String().getClass());
+        streamsProperties.put(StreamsConfig.COMMIT_INTERVAL_MS_CONFIG, 100);
+        streamsProperties.put(StreamsConfig.STATE_DIR_CONFIG, stateDirStr);
+        streamsProperties.put(StreamsConfig.NUM_STANDBY_REPLICAS_CONFIG, 1);
+
+        // it is expected that max.poll.interval, retries, request.timeout and max.block.ms set
+        // streams_broker_down_resilience_test and passed as args
+        if (additionalConfigs != null && !additionalConfigs.equalsIgnoreCase("none")) {
+            Map<String, String> updated = updatedConfigs(additionalConfigs);
+            System.out.println("Updating configs with " + updated);
+            streamsProperties.putAll(updated);
+        }
+
+        if (!confirmCorrectConfigs(streamsProperties)) {
+            System.err.println(String.format("ERROR: Did not have all required configs expected  to contain %s %s %s %s",
+                                             StreamsConfig.consumerPrefix(ConsumerConfig.MAX_POLL_INTERVAL_MS_CONFIG),
+                                             StreamsConfig.producerPrefix(ProducerConfig.RETRIES_CONFIG),
+                                             StreamsConfig.producerPrefix(ProducerConfig.REQUEST_TIMEOUT_MS_CONFIG),
+                                             StreamsConfig.producerPrefix(ProducerConfig.MAX_BLOCK_MS_CONFIG)));
+
+            System.exit(1);
+        }
+
+        final StreamsBuilder builder = new StreamsBuilder();
+
+        final KStream<String, String> streamOne = builder.stream(SOURCE_TOPIC, Consumed.with(stringSerde, stringSerde));
+
+        String inMemoryStoreName = "in-memory-store";
+        String persistentMemoryStoreName = "persistent-memory-store";
+
+        KeyValueBytesStoreSupplier inMemoryStoreSupplier = Stores.inMemoryKeyValueStore(inMemoryStoreName);
+        StoreBuilder<KeyValueStore<String, String>> storeBuilder = Stores.keyValueStoreBuilder(inMemoryStoreSupplier, stringSerde, stringSerde);
+
+        KeyValueBytesStoreSupplier persistentStoreSupplier = Stores.persistentKeyValueStore(persistentMemoryStoreName);
+        StoreBuilder<KeyValueStore<String, String>> persistentStoreBuilder = Stores.keyValueStoreBuilder(persistentStoreSupplier, stringSerde, stringSerde);
+
+        builder.addStateStore(storeBuilder);
+        builder.addStateStore(persistentStoreBuilder);
+
+        streamOne.transformValues(new StandbyTransformValuesSupplier(inMemoryStoreName), inMemoryStoreName)
```

Comment:
any reason why you don't use `builder.table(..)` instead of `transformValues(..)` ?

Reply:
not sure, but will switch